Source/ImagePickerController.swift

```diff
@@ -301,26 +312,25 @@ extension ImagePickerController: ImageGalleryPanGestureDelegate {
   func panGestureDidChange(translation: CGPoint) {
     let galleryHeight = initialFrame.height - translation.y
 
+    if galleryHeight >= GestureConstants.maximumHeight { return }
+
     if galleryHeight <= ImageGalleryView.Dimensions.galleryBarHeight {
       updateGalleryViewFrames(ImageGalleryView.Dimensions.galleryBarHeight)
-    } else if galleryHeight >= GestureConstants.maximumHeight {
-      updateGalleryViewFrames(GestureConstants.maximumHeight)
-    } else {
+
+    } else if galleryHeight >= GestureConstants.minimumHeight {
+
+      let scale = (galleryHeight - ImageGalleryView.Dimensions.galleryBarHeight) / (GestureConstants.minimumHeight - ImageGalleryView.Dimensions.galleryBarHeight)
+      galleryView.collectionView.transform = CGAffineTransformMakeScale(scale, scale)
       galleryView.frame.origin.y = initialFrame.origin.y + translation.y
       galleryView.frame.size.height = initialFrame.height - translation.y
-    }
 
-    if galleryHeight > GestureConstants.minimumHeight {
-      galleryView.collectionViewLayout.invalidateLayout()
-      galleryView.collectionView.frame.size.height = galleryView.frame.size.height - ImageGalleryView.Dimensions.galleryBarHeight
-      galleryView.collectionSize = CGSize(width: galleryView.collectionView.frame.height, height: galleryView.collectionView.frame.height)
+      let value = self.view.frame.width * (scale - 1) / scale
+      self.galleryView.collectionView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right:  value)
 
-      if galleryHeight < GestureConstants.maximumHeight {
-        let realTranslation = translation.y < -GestureConstants.minimumHeight + ImageGalleryView.Dimensions.galleryBarHeight
-          ? translation.y + GestureConstants.minimumHeight - ImageGalleryView.Dimensions.galleryBarHeight
-          : translation.y
-        galleryView.collectionView.contentOffset = CGPoint(x: initialContentOffset.x - (realTranslation * CGFloat(numberOfCells)), y: 0)
-      }
+    } else {
+
```

Comment:
I added newlines here intentionally to give more clarity. This is a math-heavy code so I decided to format it in in blocks. Try to look at the source, not the diff. Still want to remove?